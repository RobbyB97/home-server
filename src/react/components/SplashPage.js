import React from 'react'
import {NavLink} from 'react-router-dom'
import { connect } from 'react-redux'

import {splashOff} from '../actions/home'

const SplashPage = ({splash, dispatch}) => (
    <div id="splash-page" className={splash ? "":"hidden"}>
        <div class="splash-content">
            <h2 className="text-one">
                Designer. <br/> &nbsp; Developer. <br/> &nbsp;&nbsp; Problem Solver.
            </h2>
            <h3 className="text-two">
                Robby Bergers
            </h3>
            <button className="to-site" onClick={() => {
                dispatch(splashOff())
            }}>
                Splash Page button
            </button>
        </div>
    </div>
)

const mapStateToProps = (state) => {
    return {
        splash: state.home.splash
    }
  }

export default connect(mapStateToProps)(SplashPage)