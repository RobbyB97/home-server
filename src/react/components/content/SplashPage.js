import React from 'react'
import {NavLink} from 'react-router-dom'
import { connect } from 'react-redux'

import {splashOff} from '../../actions/home'

export class SplashPage extends React.Component {

    componentDidMount = () => {
        setTimeout(() => {
            let text = document.querySelector('.splash__text.one')
            text.style.opacity = '1'
        }, 200)

        setTimeout(() => {
            let text = document.querySelector('.splash__text.two')
            text.style.opacity = '1'
        }, 350)

        setTimeout(() => {
            let text = document.querySelector('.splash__text.three')
            text.style.opacity = '1'
        }, 500)

        setTimeout(() => {
            let text = document.querySelector('.splash__name')
            text.style.opacity = '1'
        }, 750)

        setTimeout(() => {
            let text = document.querySelector('.splash__to-site')
            text.style.opacity = '1'
        }, 850)
    }

    render() {
        return (
            <div id="splash" className={this.props.splash ? "":"splash__hidden"}>
                <div className="splash__background"></div>
                <div className="splash__content">
                    <p className="splash__text one">
                        Designer.
                    </p>
        
                    <p className="splash__text two">
                        Developer.
                    </p>
        
                    <p className="splash__text three">
                        Problem solver.
                    </p>
        
                    <p className="splash__name">
                        Robby Bergers.
                    </p>
                    <button className="splash__to-site" onClick={() => {
                        this.props.dispatch(splashOff())
                    }}>
                        Learn more â†’
                    </button>
                </div>
            </div> 
        )
    }
}

const mapStateToProps = (state) => {
    return {
        splash: state.home.splash
    }
  }

export default connect(mapStateToProps)(SplashPage)