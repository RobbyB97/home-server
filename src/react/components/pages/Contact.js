import React from 'react'
import {connect} from 'react-redux'



const Contact = ({dispatch, nav, title}) => (
	<div id="contact">
		Contact component
	</div>
)



const mapStateToProps = (state) => {
	return {
		nav: state.navigation.nav,
		title: state.title.title,
	}
}

export default connect(mapStateToProps)(Contact)