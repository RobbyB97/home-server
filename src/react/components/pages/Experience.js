import React from 'react'
import {connect} from 'react-redux'

import Job from '../content/Job'
import PrevPage from '../common/PrevPage'
import NextPage from '../common/NextPage'


export class Experience extends React.Component {
    
    render() {
        return (
            <div id="experience">
                
                Experience component
                <Job
                    title="Finalsite"
                    text="Job"
                    image="/dist/images/icons/finalsite.png">
                </Job>

                <PrevPage
                    text="Skills"
                    route="/skills/"
                    float={true}
                />

                <NextPage 
                    text="Projects"
                    route="/projects/"
                    float={true}
                />
            </div>  
        )
    }
}


const mapStateToProps = (state) => {
	return {
		nav: state.navigation.nav,
        title: state.title.title,
        splash: state.home.splash
	}
}

const mapDispatchToProps = (dispatch) => ({

})


export default connect(mapStateToProps, mapDispatchToProps)(Experience)