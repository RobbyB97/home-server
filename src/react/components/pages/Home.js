import React from 'react'
import {connect} from 'react-redux'

import Attribute from '../content/Attribute'


export class Home extends React.Component {

    render() {

        const fastAttribute = {
            image: "/dist/images/icons/speed_icon.png",
            title: "Fast",
            text: "Fully minified files and an optimized structure guarantee your website will load quickly, even on inconsistent internet connections."
        }

        const responsiveAttribute = {
            image: "/dist/images/icons/responsive_icon.png",
            title: "Responsive",
            text: "Mobile first development built up to 4k screens guarantees that your website will be elegant and functional on any device."
        }

        const beautifulAttribute = {
            image: "/dist/images/icons/paint_icon.png",
            title: "Beautiful",
            text: "A thorough, UX focused design process and open communication guarantees your website will look as beautiful as you envision it."
        }

        const accessibleAttribute = {
            image: "/dist/images/icons/accessibility_icon.png",
            title: "Accessible",
            text: "Full compliance with ADA standards guarantees that your website will be accessible and easy to use for everyone."
        }
        
        return (
            <div id="home">
                
                <div className={this.props.splash ? "home__background phone one": "home__background phone two"}></div>
                <div className={this.props.splash ? "home__background tablet one": "home__background tablet two"}></div>
                <div className={this.props.splash ? "home__background desktop one": "home__background desktop two"}></div>
                <div className={this.props.splash ? "home__background huge one": "home__background huge two"}></div>

                {!this.props.splash &&
                    <div className="home__attributes">
                        <Attribute {...fastAttribute} />
                        <Attribute {...responsiveAttribute} />
                        <Attribute {...beautifulAttribute} />
                        <Attribute {...accessibleAttribute} />
                    </div>
                }
                
            </div>
        )
    }
}


const mapStateToProps = (state) => {
    return {
      nav: state.navigation.nav,
      title: state.title.title,
      home: state.home.home,
      homePage: state.home.homePage,
      splash: state.home.splash
    }
}

const mapDispatchToProps = (dispatch) => ({

})


export default connect(mapStateToProps, mapDispatchToProps)(Home)