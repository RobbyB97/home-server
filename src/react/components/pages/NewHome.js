import React from 'react'


export class NewHome extends React.Component {
    render() {
        return (
            <div id="newHome">                
                <div className="newHome__content">
                    <div className="newHome__head">
                        <div className="newHome__name">
                            <h2>Robby Bergers</h2>
                        </div>
                        <div className="newHome__title">
                            <h3>Front End Developer</h3>
                        </div>
                    </div>

                    <div className="newHome__body">

                    </div>
                </div>

                <div className="newHome__backdrop"></div>
            </div>
        )
    }
}


export default NewHome