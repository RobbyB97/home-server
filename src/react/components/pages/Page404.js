import React from 'react'
import {NavLink} from 'react-router-dom'


export class Page404 extends React.Component {
    
    render() {
        return (
            <div id="page404">
                <div className="page404__backdrop"></div>
                <div className="page404__content">
                    <div className="page404__image">
                        <img src="/dist/images/wildcard.png" />
                    </div>

                    <div className="page404__text">
                        Uh oh, looks like this page doesn't exist.
                    </div>

                    <div className="page404__home-button">
                        <NavLink to="/">
                            <button>
                                Return home
                            </button>
                        </NavLink>
                    </div>
                </div>
            </div>
        )
    }
}


export default Page404