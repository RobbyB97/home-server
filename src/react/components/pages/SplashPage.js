import React from 'react'
import {NavLink} from 'react-router-dom'
import { connect } from 'react-redux'

import {splashOff} from '../../actions/home'

export class SplashPage extends React.Component {
    
    componentDidMount = () => {
        this.splashFadeIn()
    }

    splashOff = () => {
        this.props.splashOff()
    }

    splashFadeIn = () => {
        setTimeout(() => {
            const background = document.querySelector('.splash__background.two')
            background.style.opacity = '1'
        }, 2900)

        setTimeout(() => {
            const text = document.querySelector('.splash__words.one')
            text.style.opacity = '1'
        }, 200)
        
        setTimeout(() => {
            const text = document.querySelector('.splash__words.two')
            text.style.opacity = '1'
        }, 800)

        setTimeout(() => {
            const text = document.querySelector('.splash__words.three')
            text.style.opacity = '1'
        }, 1100)

        setTimeout(() => {
            const text = document.querySelector('.splash__words.four')
            text.style.opacity = '1'
        }, 1400)

        setTimeout(() => {
            const text = document.querySelector('.splash__words.five')
            text.style.opacity = '1'
        }, 1700)

        setTimeout(() => {
            const text = document.querySelector('.splash__words.six')
            text.style.opacity = '1'
        }, 2000)

        setTimeout(() => {
            const text = document.querySelector('.splash__words.seven')
            text.style.opacity = '1'
        }, 2300)

        setTimeout(() => {
            const text = document.querySelector('.splash__words.eight')
            text.style.opacity = '1'
        }, 2900)

        setTimeout(() => {
            const text = document.querySelector('.splash__words.nine')
            text.style.opacity = '1'
        }, 3500)

        setTimeout(() => {
            let text = document.querySelector('.splash__name')
            text.style.opacity = '1'
            text = document.querySelector('.splash__to-site')
            text.style.opacity = '1'
        }, 3900)
    }

    render() {
        return (
            <div id="splash" className={this.props.splash ? "":"splash__hidden"}>
                <div className="splash__background one"></div>
                <div className="splash__background two"></div>

                <div className="splash__content">
                    <p className="splash__text one">
                        <span className="splash__words one">
                            What is a perfect website?&nbsp;
                        </span>
                        <span className="splash__words two">
                            The answer is a&nbsp;
                        </span>
                        <span className="splash__words three">
                            <i>memorable</i>,&nbsp;
                        </span>
                        <span className="splash__words four">
                            <i>simplistic</i>,&nbsp;
                        </span>
                        <span className="splash__words five">
                            <i>intuitive</i>,&nbsp;
                        </span>
                        <span className="splash__words six">
                            and <i>fast</i>&nbsp;
                        </span>
                        <span className="splash__words seven">
                            user experience.&nbsp;
                        </span>
                        <span className="splash__words eight">
                            My goal is to create that perfect website for your business.
                        </span>
                    </p>

                    <p className="splash__text two">
                        <span className="splash__words nine">
                            Hi, my name's Robby. I'm a freelance front end developer. If you've been searching for someone to design an elegant website or web app that your clients will love, your search is over.
                        </span>
                    </p>
        
                    <p className="splash__name">
                        Robby Bergers.
                    </p>
                    <button className="splash__to-site" onClick={() => {
                        this.splashOff()
                    }}>
                        Learn more â†’
                    </button>
                </div>
            </div> 
        )
    }
}


const mapStateToProps = (state) => {
    return {
        splash: state.home.splash
    }
}

const mapDispatchToProps = (dispatch) => ({
    splashOff: () => dispatch(splashOff())
})


export default connect(mapStateToProps, mapDispatchToProps)(SplashPage)