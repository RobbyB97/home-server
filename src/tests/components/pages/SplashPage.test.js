import React from 'react'
import {shallow} from 'enzyme'

import {SplashPage} from '../../../react/components/pages/SplashPage'


test('Render SplashPage', () => {
    const wrapper = shallow(<SplashPage test={true}/>)

    expect(wrapper).toMatchSnapshot()
})


test('Trigger splashOff', () => {
    const splashOff = jest.fn()
    const wrapper = shallow(
        <SplashPage 
            test={true}
            splashOff={splashOff}    
        />
    )
    wrapper.find('.splash__to-site').simulate('click')

    expect(splashOff).toHaveBeenCalled()
})