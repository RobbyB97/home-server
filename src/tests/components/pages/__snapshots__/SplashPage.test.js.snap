// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render SplashPage 1`] = `
<div
  className="splash__hidden"
  id="splash"
>
  <div
    className="splash__background one"
  />
  <div
    className="splash__background two"
  />
  <div
    className="splash__content"
  >
    <p
      className="splash__text one"
    >
      <span
        className="splash__words one"
      >
        What is a perfect website? 
      </span>
      <span
        className="splash__words two"
      >
        The answer is a 
      </span>
      <span
        className="splash__words three"
      >
        <i>
          memorable
        </i>
        , 
      </span>
      <span
        className="splash__words four"
      >
        <i>
          simplistic
        </i>
        , 
      </span>
      <span
        className="splash__words five"
      >
        <i>
          intuitive
        </i>
        , 
      </span>
      <span
        className="splash__words six"
      >
        and 
        <i>
          fast
        </i>
         
      </span>
      <span
        className="splash__words seven"
      >
        user experience. 
      </span>
      <span
        className="splash__words eight"
      >
        My goal is to create that perfect website for your business.
      </span>
    </p>
    <p
      className="splash__text two"
    >
      <span
        className="splash__words nine"
      >
        Hi, my name's Robby. I'm a freelance front end developer. If you've been searching for someone to design an elegant website or web app that your clients will love, your search is over.
      </span>
    </p>
    <p
      className="splash__name"
    >
      Robby Bergers.
    </p>
    <button
      className="splash__to-site"
      onClick={[Function]}
    >
      Learn more →
    </button>
  </div>
</div>
`;
